control/state.py

```diff
@@ -350,34 +352,79 @@ def check_for_old_format_omap_files(self):
            if omap_item_key.startswith("bdev"):
                raise Exception("Old OMAP file format, still contains bdevs, please remove file and try again")
 
-    def fetch_and_display_ceph_version(self, conn):
+    def fetch_and_display_ceph_version(self):
         try:
-            rply = conn.mon_command('{"prefix":"mon versions"}', b'')
+            rply = self.conn.mon_command('{"prefix":"mon versions"}', b'')
             ceph_ver = rply[1].decode().removeprefix("{").strip().split(":")[0].removeprefix('"').removesuffix('"')
             ceph_ver = ceph_ver.removeprefix("ceph version ")
             self.logger.info(f"Connected to Ceph with version \"{ceph_ver}\"")
         except Exception as ex:
             self.logger.debug(f"Got exception trying to fetch Ceph version: {ex}")
             pass
 
-    def fetch_ceph_fsid(self, conn) -> str:
+    def fetch_ceph_fsid(self) -> str:
         fsid = None
         try:
-            fsid = conn.get_fsid()
+            fsid = self.conn.get_fsid()
         except Exception as ex:
             self.logger.debug(f"Got exception trying to fetch Ceph fsid: {ex}")
             pass
 
         return fsid
 
+    def pool_exists(self, pool) -> bool:
```

Comment:
nit: code `self.gateway_state.omap.pool_exists(rbd_pool_name)` looks strange, since this operation is not related to omap/state. could we consider extracting pool, fsid , image logic into a separate class/file?

Reply:
@baum thought about it but used the omap code as I had to use the rados connection which is there and I didn't want to make too many changes just for this.

Reply:
I realize that before the change the only part of the code that used the RADOS connection internally was the OMAP state, however not sure that the new functionality of checking pool existence or creating images should go under OMAP state 😌

Reply:
I agree with @baum , it should not go under OMAP state.

Reply:
Done